### PR TITLE
ContextHandler: add all configured auth header to context

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
When using authn.Service to perform authentication in context handler we need to add auth headers to the list stored in config.

Instead of checking what method was used I add all possible headers that can be used by checking config.
We could possible build this list when initiating context handler so we don't have to rebuilt it on every request

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Jguer — APPROVED** [review]

File: pkg/services/contexthandler/contexthandler.go [hunk 1]
```diff
@@ -742,6 +742,32 @@ type authHTTPHeaderListContextKey struct{}
...
+	if list == nil {
+		list = &AuthHTTPHeaderList{
+			Items: []string{},
+		}
+	}
+
+	list.Items = append(list.Items, "Authorization")
+
+	if cfg.JWTAuthEnabled && cfg.JWTAuthHeaderName != "" {
+		list.Items = append(list.Items, cfg.JWTAuthHeaderName)
```
Comment (Jguer) [inline pkg/services/contexthandler/contexthandler.go:756]: These can also be `Authorization` but I think it won't lead to side effects in the current implementation

**Review by marefr — COMMENTED** [review]
Think this would work. Think you can remove the `WithAuthHTTPHeader` and the usage of it as well?

**Comment by kalleep** [comment]
@marefr Changed so that all possible auth headers are added to the list for every request. Had to update some tests because I removed the old function

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**